xl-v1-object.go

```diff
@@ -368,20 +372,28 @@ func (xl xlObjects) PutObject(bucket string, object string, size int64, data io.
 	minioMetaTmpBucket := path.Join(minioMetaBucket, tmpMetaPrefix)
 	tempObj := uniqueID
 
-	// Lock the object.
-	nsMutex.Lock(bucket, object)
-	defer nsMutex.Unlock(bucket, object)
-
-	// Initialize xl meta.
-	xlMeta := newXLMetaV1(object, xl.dataBlocks, xl.parityBlocks)
-
+	var partsMetadata []xlMetaV1
+	var errs []error
+	nsMutex.RLock(bucket, object)
 	// Read metadata associated with the object from all disks.
-	partsMetadata, errs := readAllXLMetadata(xl.storageDisks, bucket, object)
+	partsMetadata, errs = readAllXLMetadata(xl.storageDisks, bucket, object)
```

Comment:
you could use previous style `partsMetadata, errs := readAllXLMetadata(xl.storageDisks, bucket, object)` right?